src/enhancers/withState.js

```diff
@@ -1,18 +1,26 @@
-import {useState} from 'react';
+import {useState, useMemo} from 'react';
+import {PROPS} from '../util/constants';
 
-export default (stateName, stateUpdaterName, initialValue) => ({
+export const withState = (stateName, stateUpdaterName, initialValue) => ({
   generateNewVariable,
 }) => {
   const initialValueAlias = generateNewVariable();
+  const initialValueIsFunctionAlias = generateNewVariable();
+  const computedInitialValueAlias = generateNewVariable();
   const stateValue = generateNewVariable();
 
   return {
     dependencies: {
       useState,
+      useMemo,
       [initialValueAlias]: initialValue,
+      [initialValueIsFunctionAlias]: Object.prototype.toString.call(initialValue) === '[object Function]',
```

Comment:
why not use typeof ?

Reply:
I grabbed this implementation from lodash. This wasn't the article I read (http://es-discourse.com/t/why-typeof-is-no-longer-safe/15), but I got deep into MDN and a stack overflow discussion about some inconsistent behavior with typeof. It probably wouldn't happen in our limited case, but I felt safe with whatever lodash did...

Reply:
Okay I updated `withState` let me know what you think.